src/Futhark/Analysis/LastUse.hs

```diff
@@ -0,0 +1,164 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | Provides last-use analysis for Futhark programs.
+module Futhark.Analysis.LastUse (LastUseMap, analyseProg) where
+
+import Control.Arrow (first)
```

Comment:
Don't import `Control.Arrow` to get this; use `Data.Bifunctor` (which also has similar goodies without the conceptual overhead of arrows).